scripts/deployContracts.js

```diff
@@ -0,0 +1,186 @@
+const { Contracts, ZWeb3 } = require('@openzeppelin/upgrades');
+const HDWalletProvider = require('truffle-hdwallet-provider')
+const Web3 = require('web3');
+const moment = require('moment');
+const args = process.argv;
+require('dotenv').config();
+
+// Get network to use from arguments
+let network = 'mainnet', mnemonic, provider, web3;
+for (var i = 0; i < args.length; i++) {
+  if (args[i] == '--network')
+    network = args[i+1];
+  if (args[i] == '--provider')
+    provider = args[i+1];
+}
+if (!provider) throw('Not provider selected, --provider parameter missing');
+
+mnemonic = process.env.REACT_APP_KEY_MNEMONIC;
+
+console.log('Running deploy on network', network, 'using http provider', provider)
+const web3Provider = new HDWalletProvider(mnemonic, new Web3.providers.HttpProvider(provider), 0, 10);
+web3 = new Web3(web3Provider)
+
+ZWeb3.initialize(web3.currentProvider);
+Contracts.setLocalBuildDir('contracts/build/');
+
+const DATContract = Contracts.getFromLocal('DecentralizedAutonomousTrust');
+const AdminUpgradeabilityProxy = Contracts.getFromLocal('AdminUpgradeabilityProxy');
+const ProxyAdmin = Contracts.getFromLocal('ProxyAdmin');
+const TokenVesting = Contracts.getFromLocal('TokenVesting');
+const contracts = {};
+
+const momentNow = moment.utc((new Date()).toUTCString());
+
+// Deploy config taken from
+// https://daotalk.org/t/configuration-template-for-fundraising-decentralized-application/1250
+// https://alchemy.daostack.io/dao/0x519b70055af55a007110b4ff99b0ea33071c720a/proposal/0xeb9cf2b3d76664dc1e983137f33b2400ad11966b1d79399d7ca55c25ad6283fa
+
+const pricePerDXDUSD = 25;
+const pricePerETHUSD = 200;
+const initGoalUSD = 50000;
+const initiGoalDXD = initGoalUSD/pricePerDXDUSD;
+const initGoalETH = initGoalUSD/pricePerETHUSD;
+
+// From DAT conract:
+// tokenValue = BigDiv.bigDiv2x1(
+//   currencyValue,
+//   2 * buySlopeDen,
+//   initGoal * buySlopeNum
+// );
+const buySplopeDen = ((initiGoalDXD * (initiGoalDXD*1)) / initGoalETH) / 2;
```

Comment:
@jpkcambridge check this. I tested it. It uses buySlopDen as x based in the calculation made in the DAT contract. I tested it locally and I works.